api/stse_device_authentication.c

```diff
@@ -1,182 +1,216 @@
+ * the root directory of this software component.
+ * If no LICENSE file comes with this software, it is provided AS-IS.
+ *
+ *****************************************************************************/
+
+/* Includes ------------------------------------------------------------------*/
+
+#include <api/stse_ecc.h>
+#include <api/stse_device_authentication.h>
+
+#include "certificate/stse_certificate.h"
+#include "certificate/stse_certificate_subparsing.h"
+#include "certificate/stse_certificate_crypto.h"
+
+#define CERTIFICATE_SIZE_OFFSET_BYTES 	2U
+#define CERTIFICATE_SIZE_LENGTH 		2U
+#define CERTIFICATE_OFFSET_BYTES		0U
+#define CERTIFICATE_DEVICE_ID_OFFSET	15U
+#define CERTIFICATE_DEVICE_ID_SIZE		11U
+
+/* Static functions declaration ----------------------------------------------*/
+
+/* Exported functions --------------------------------------------------------*/
+
+stse_ReturnCode_t stse_get_device_id(stse_Handler_t * pSTSE, PLAT_UI8 Certificate_zone, PLAT_UI8* device_id)
+{
+	volatile stse_ReturnCode_t ret = STSE_API_INVALID_PARAMETER;
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+
+	ret = stse_data_storage_read_data_zone(
+			pSTSE,							/* STSAFE handler */
+			Certificate_zone,				/* Certificate zone */
+			CERTIFICATE_DEVICE_ID_OFFSET,	/* X bytes offset */
+			device_id,						/* Returned certificate size */
+			CERTIFICATE_DEVICE_ID_SIZE,		/* Certificate size length */
+			0,								/* No maximum chunck size (No chunck at all) */
+			STSAFEA_NO_PROT);				/* No protection */
+
+	return ret;
+}
+
+stse_ReturnCode_t stse_get_device_certificate_size(stse_Handler_t * pSTSE, PLAT_UI8 Certificate_zone, PLAT_UI16 * certificate_size)
+{
+	volatile stse_ReturnCode_t ret = STSE_API_INVALID_PARAMETER;
+	PLAT_UI8 certificate_size_ui8[2];
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+
+
+	ret = stse_data_storage_read_data_zone(
+			pSTSE,							/* STSAFE handler */
+			Certificate_zone,				/* Certificate zone */
+			CERTIFICATE_SIZE_OFFSET_BYTES,	/* 2 bytes offset */
+			certificate_size_ui8,			/* Returned certificate size */
+			CERTIFICATE_SIZE_LENGTH,		/* Certificate size length 2 bytes */
+			0,								/* No maximum chunck size (No chunck at all) */
+			STSAFEA_NO_PROT);				/* No protection */
+
+	if(ret != STSE_OK)
+	{
+		return( ret );
+	}
+
+	*certificate_size = (((uint16_t)certificate_size_ui8[0]) << 8) + ((uint16_t)certificate_size_ui8[1]) + 4U;
+
+
+	return STSE_OK;
+}
+
+stse_ReturnCode_t stse_get_device_certificate(stse_Handler_t * pSTSE, PLAT_UI8 Certificate_zone, PLAT_UI16 certificate_size, PLAT_UI8* stsafe_certificate)
+{
+	volatile stse_ReturnCode_t ret = STSE_API_INVALID_PARAMETER;
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+

```

Comment:
- Add a check on stsafe_certificate to avoid NULL pointer
- Rename `stsafe_certificate` in `pStse_certificate`

```C
	if (pStse_certificate == NULL)
	{
		return( STSE_API_INVALID_PARAMETER );
	}
```